python_modules/dagster/dagster_tests/components_tests/unit_tests/test_unresolvable_component.py

```diff
@@ -0,0 +1,27 @@
+import dagster as dg
+import pytest
+from dagster.components.core.tree import ComponentTreeException
+from dagster.components.testing.utils import create_defs_folder_sandbox
+
+
+class UnresolvableComponent(dg.Component, dg.Model):
+    """This component class does not subclass Resolvable."""
+
+    some_field: str
+
+    def build_defs(self, context: dg.ComponentLoadContext) -> dg.Definitions: ...
+
+
+def test_unresolvable_component():
+    with create_defs_folder_sandbox() as sandbox:
+        defs_path = sandbox.scaffold_component(
+            UnresolvableComponent,
+            defs_yaml_contents={
+                "type": "dagster_tests.components_tests.unit_tests.test_unresolvable_component.UnresolvableComponent",
+                # this component is not resolvable and so cannot have attributes
+                "attributes": {"some_field": "foo"},
+            },
+        )
+        with pytest.raises(ComponentTreeException):
```

Comment:
Does the exception point you at inheriting `Resolvable` or explain the situation effectively? 

This example where its a Model with basic field types seems like maybe it should work out of the box

Reply:
maybe it should work out of the box, although then you get a rough edge at a different point where if you add a complex field to something that previously only supported strings, things start breaking at that point instead of from the start